src/librustc/ty/print/pretty.rs

```diff
@@ -1425,6 +1492,19 @@ impl<F: fmt::Write> PrettyPrinter<'tcx> for FmtPrinter<'_, 'tcx, F> {
             ty::ReClosureBound(_) => true,
         }
     }
+
+    fn pretty_print_const_pointer(
```

Comment:
So is this used both for MIR dumps and something user-visible? Or why do we only *sometimes* print the AllocId?

Reply:
Yes, the pretty printer is used for both MIR dumps and error messages, and the "show alloc id" flag is only set by MIR dumps

Reply:
But there's also something it is *not* used for and that goes through the other `pretty_print_const_pointer`?

Reply:
Yes, the `SymbolPrinter` and the `AbsolutePathPrinter`. The latter is used for `type_name`, which we probably don't want to make users see the alloc ids.

`SymbolPrinter`... May actually need the ids. cc @eddyb

Reply:
> `SymbolPrinter`... May actually need the ids. cc @eddyb

Which one? `legacy` symbols don't matter much, the hash contains all of the information, they should just be consistent across crates. You don't have to show the const's value if you can't do it consistently cross-crate.

`v0` ones are trickier, and I feel like the push for `#[structural_match]` in `const` generics is going to force us to implement some sort of general ADT representation for constants in `v0` symbols, sooner than later.
(it'd likely be similar to how all values of `#[structural_match]` types can technically be turned into some sort of pattern)

cc @michaelwoerister @varkor @yodaldevoid

Reply:
For reference, the dedicated issue for the last comment is https://github.com/rust-lang/rust/issues/61486.

Reply:
I could set it to print just integer types for now and `bug!` out for other types so we don't silently introduce bugs